pageserver/src/tenant/secondary.rs

```diff
@@ -250,7 +211,10 @@ impl SecondaryTenant {
             // of the cache.
             let mut detail = this.detail.lock().unwrap();
             if let Some(timeline_detail) = detail.timelines.get_mut(&timeline_id) {
-                timeline_detail.on_disk_layers.remove(&name);
+                let removed = timeline_detail.on_disk_layers.remove(&name);
+                if let Some(removed) = removed {
+                    removed.remove_blocking();
+                }
```

Comment:
The previous code seemed to skip updating some in-memory state if the file was not present. Do we not need that?

Reply:
Hmm... yes, we should pull the evicted_at update into this conditional block.  It doesn't break anything if we don't, just leaves some chaff in evicted_at in the case of a race between eviction and deletions during download.

Reply:
4c516faa7